Split/Storage/Events/EventDao.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  EventDao.swift
+//  Split
+//
+//  Created by Javier L. Avrudsky on 12/11/2020.
+//  Copyright © 2020 Split. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+protocol EventDao {
+    func insert(_ event: EventDTO)
+    func getBy(createdAt: Int64, status: Int32, maxRows: Int) -> [EventDTO]
+    func update(ids: [String], newStatus: Int32)
+    func delete(_ events: [EventDTO])
+}
+
+class CoreDataEventDao: EventDao {
+
+    let coreDataHelper: CoreDataHelper
+
+    init(coreDataHelper: CoreDataHelper) {
+        self.coreDataHelper = coreDataHelper
+    }
+
+    func insert(_ event: EventDTO) {
+        if let obj = coreDataHelper.create(entity: .event) as? EventEntity {
+            do {
+                obj.storageId = coreDataHelper.generateId()
+                obj.body = try Json.dynamicEncodeToJson(event)
+                obj.createdAt = Date().unixTimestamp()
+                obj.status = StorageRecordStatus.active
+                coreDataHelper.save()
+            } catch {
+                Logger.e("An error occurred while loading events from storage: \(error.localizedDescription)")
```

Comment:
since the method is `insert`, should the log be something like "An error occurred while saving events to storage: \(error.localizedDescription)" ?

Reply:
Oh yes! Thanks doc!